server/routes/index.ts

```diff
@@ -74,6 +78,66 @@ router.get('/health', async (req, res) => {
   res.send('OK').end()
 })
 
+const substackDomainCache = {
+  // uncomment the below for test purpose
+  'localhost:3100': 'polymorpher.substack.com'
+}
+
+const client = buildClient()
+
+router.get('/substack/api/v1/archive',
+  limiter(),
+  async (req, res) => {
+    const host = req.get('host')
+
+    if (!host) {
+      return
+    }
+
+    let substackDomain = substackDomainCache[host]
+
+    if (!substackDomain) {
+      const [subdomain, sld] = host?.split('.')
```

Comment:
need to use getSld from util to handle some special domain mapping. I will fix

Reply:
Done